internal/kldcontracts/smartcontractgw.go

```diff
@@ -829,24 +834,35 @@ func (g *smartContractGW) resolveAddressOrName(id string) (deployMsg *kldmessage
 	return deployMsg, registeredName, info, err
 }
 
-func (g *smartContractGW) isSwaggerRequest(req *http.Request) (swaggerRequest, uiRequest, factoryOnly, abiRequest bool, from string) {
+func (g *smartContractGW) isSwaggerRequest(req *http.Request) (swaggerGen *kldopenapi.ABI2Swagger, uiRequest, factoryOnly, abiRequest bool, from string) {
 	req.ParseForm()
+	var swaggerRequest bool
 	if vs := req.Form["swagger"]; len(vs) > 0 {
-		swaggerRequest = true
+		swaggerRequest = strings.ToLower(vs[0]) != "false"
 	}
 	if vs := req.Form["openapi"]; len(vs) > 0 {
-		swaggerRequest = true
+		swaggerRequest = strings.ToLower(vs[0]) != "false"
 	}
 	if vs := req.Form["ui"]; len(vs) > 0 {
-		uiRequest = true
+		uiRequest = strings.ToLower(vs[0]) != "false"
 	}
 	if vs := req.Form["factory"]; len(vs) > 0 {
-		factoryOnly = true
+		factoryOnly = strings.ToLower(vs[0]) != "false"
 	}
 	if vs := req.Form["abi"]; len(vs) > 0 {
-		abiRequest = true
+		abiRequest = strings.ToLower(vs[0]) != "false"
 	}
 	from = req.FormValue("from")
+	if swaggerRequest {
+		var conf = *g.baseSwaggerConf
+		if vs := req.Form["noauth"]; len(vs) > 0 {
+			conf.BasicAuth = strings.ToLower(vs[0]) == "false"
+		}
+		if vs := req.Form["schemes"]; len(vs) > 0 {
+			conf.ExternalSchemes = strings.Split(vs[0], ",")
```

Comment:
Should we filter out any schemes other than `http/https` here, with a warning?

Reply:
good point - injection potential